pandas/tests/generic/test_generic.py

```diff
@@ -83,21 +83,21 @@ def test_get_numeric_data(self, frame_or_series):
         }
 
         # get the numeric data
-        o = construct(n, **kwargs)
+        o = construct(frame_or_series, n, **kwargs)
         result = o._get_numeric_data()
         tm.assert_equal(result, o)
 
         # non-inclusion
         result = o._get_bool_data()
-        expected = construct(n, value="empty", **kwargs)
+        expected = construct(frame_or_series, n, value="empty", **kwargs)
         if isinstance(o, DataFrame):
             # preserve columns dtype
             expected.columns = o.columns[:0]
-        tm.assert_equal(result, expected)
+        tm.assert_equal(result.reset_index(drop=True), expected)
```

Comment:
@topper-123 is this right?

```python
In [1]: ser = Series([1,2,3])

In [2]: ser.index
Out[2]: RangeIndex(start=0, stop=3, step=1)

In [3]: ser._get_bool_data()
Out[3]: Series([], dtype: int64)

In [4]: ser._get_bool_data().index
Out[4]: Index([], dtype='object')
```

Reply:
@phershbe OK for now could you please just add a comment with
```
# https://github.com/pandas-dev/pandas/issues/50862
```
above this line? then we can get this in and address that separately if needed